avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -897,46 +914,22 @@ private void processBytes(JBlock body, FieldAction action, BiConsumer<JBlock, JE
   private void processString(Schema schema, JBlock body, FieldAction action,
       BiConsumer<JBlock, JExpression> putValueIntoParent, Supplier<JExpression> reuseSupplier) {
     if (action.getShouldRead()) {
-      if (reuseSupplier.get().equals(JExpr._null())) {
-        JExpression readStringInvocation;
-
-        if (schema.getType().equals(Schema.Type.STRING) && !useGenericTypes && SchemaAssistant.isStringable(schema)) {
-          readStringInvocation = JExpr._new(schemaAssistant.classFromSchema(schema))
-              .arg(JExpr.invoke(JExpr.direct(DECODER), "readString").arg(JExpr._null()).
-                  invoke("toString"));
+      JExpression readStringInvocation;
+      JClass stringClass = schemaAssistant.findStringClass(schema);
+      if (stringClass.equals(codeModel.ref(Utf8.class)))  {
```

Comment:
there's an assumption being made here that is the schema says property Record.f should be a string any generated class Record.class "respects" that. this assumption does not generally hold - class Record could have been generated under avro < 1.8 (even if the schema itself has modern annotations)